pkg/api/ds_query_test.go

```diff
@@ -249,7 +249,7 @@ func TestDataSourceQueryError(t *testing.T) {
 			expectedStatus: errutil.StatusInternal.HTTPStatus(),
 			expectedBody: body{
 				Message:    "An error occurred within the plugin",
-				MessageId:  "plugin.downstreamError",
+				MessageId:  "plugin.requestError",
```

Comment:
I think the issue with substituting `downstream` with `request` makes me immediately think of client side errors (IE HTTP 4xx scenarios). What about `plugin.requestFailureError`? I realize failure and error are almost synonymous terms, but request failure is more specific and "error" is here by convention.

Reply:
I don't think you are overthinking it! Your arguments make sense and I agree that `plugin.requestFailureError` is better. I'll update the PR.